src/lib.rs

```diff
@@ -1062,6 +1077,42 @@ where
         self.oauth2_client.exchange_code(code)
     }
 
+    ///
+    /// Creates a request builder for exchanging a device code for an access token.
+    ///
+    /// See https://tools.ietf.org/html/rfc8628#section-3.4
+    ///
+    pub fn exchange_device_code(
+        &self,
+    ) -> Result<DeviceAuthorizationRequest<TE>, ConfigurationError> {
+        let request = self.oauth2_client.exchange_device_code();
+        if self.use_openid_scope {
+            Ok(request?.add_scope(Scope::new(OPENID_SCOPE.to_string())))
+        } else {
+            request
+        }
+    }
+
+    ///
+    /// Creates a request builder for exchanging a device code for an access token.
+    /// This method is similar to [`exchange_device_code`][Client::exchange_device_code] but
+    /// allows the caller to specify additional parameters to be sent to the authorization server.
+    /// This is useful for providers that require additional parameters to be sent in the request
+    /// body.
+    ///
+    /// See https://tools.ietf.org/html/rfc8628#section-3.4
+    ///
+    pub fn exchange_device_token<'a, 'b, 'c>(
```

Comment:
let's rename to `exchange_device_access_token` for consistency with `oauth2` and have it follow the same type signature (with generic `EF` so that users can access any vendor-specific fields that providers add). We'll need to re-export the `ExtraDeviceAuthorizationFields` trait above too (in addition to the `EmptyExtraDeviceAuthorizationFields` struct).